File: src/main/java/tn/esprit/piproject/Entities/Notification.java
package tn.esprit.piproject.Entities;
import com.mongodb.lang.Nullable;
import jakarta.persistence.Id;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document(collection = "notifications")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Notification {
    @Id
    private int id;
    private String message;
    private Date timestamp;
    private NotificationStatus status;
    @DBRef
    @Nullable
    private User sender;
    @DBRef
    @Nullable
    private User recipient;
}

File: AngularPIFront/src/app/core/Notification.ts
import { User } from "./User";

export enum NotificationStatus {
    READ = 'read',
    UNREAD = 'unread'
  }
  
  export interface Notification {
    id: number;
    sender: User;
    recipient: User;
    message: string;
    timestamp: Date;
    status: NotificationStatus;
  }